parser/ParseBlockProperties.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ *   Copyright 2011-2015 Quickstep Technologies LLC.
+ *   Copyright 2015 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "parser/ParseBlockProperties.hpp"
+
+#include <string>
+ 
+namespace quickstep {
+
+const std::string ParseBlockPropertyItem::AllString = std::string("ALL");
+
+std::string ParseBlockPropertyItem::getPropertyString() const {
+  switch(property_) {
+    case kCompress:
+      return "compress";
+    case kSort:
+      return "sort";
+    case kType:
+      return "type";
+    default:
+      return "unknown";
+  }
+}
+
+bool ParseBlockPropertyItem::compressAll() const {
+  return (property_ == kCompress && values_->begin()->value().compare(AllString) == 0);
```

Comment:
What if there are multiple `values_` and one of them (not the first) is ALL? What if the user specifies a column named "ALL" with double-quotes (so that the lexer/parser treat it as a name literal and not the SQL ALL keyword)?

Rather than doing late string-comparison here, I think it makes more sense to have a special boolean flag indicating "all" that is set by a special constructor for ParseBlockPropertyItem. Then, the parser could create the ParseBlockPropertyItem with that constructor when it encounters the SQL keyword ALL by itself, but it will not become confused in other circumstances (e.g. if double-quoted "ALL" is used as a column name).